enterprise/app/repo/repo.tsx

```diff
@@ -239,6 +304,16 @@ export default class RepoComponent extends React.Component<RepoComponentProps, R
   }
 
   async handleDeployClicked(repoResponse: repo.CreateRepoResponse) {
+    let isGCPDeploy = this.getSecrets().includes(gcpRefreshTokenKey);
+    let needsGCPLink =
+      isGCPDeploy && !this.state.secretsResponse?.secret.map((s) => s.name).includes(gcpRefreshTokenKey);
+
+    if (isGCPDeploy && needsGCPLink) {
+      await this.linkGoogleCloud()
```

Comment:
if there's an error then it will continue due to the `catch` - is that OK here too?